packages/expo-image-picker/src/ImagePicker.types.ts

```diff
@@ -295,18 +298,105 @@ export type ImagePickerErrorResult = {
 };
 
 // @needsAudit
+export type ImagePickerResult = {
+  /**
+   * An array of picked assets or `null` when the request was canceled.
+   */
+  assets: ImagePickerAsset[] | null;
+  /**
+   * Boolean flag which shows if request was canceled. If asset data have been returned this should
+   * always be `false`.
+   */
+  canceled: boolean;
+  /**
+   * @hidden
+   * @deprecated Use an American English spelling: `canceled`.
+   */
+  cancelled?: boolean;
+  /**
+   * @hidden
+   * @deprecated This field is deprecated and will be removed in SDK 48, you can access selected assets through the "assets" array instead.
+   */
+  selected?: ImagePickerAsset[];
+  /**
+   * @hidden
+   * @deprecated This field is deprecated and will be removed in SDK 48, you can access selected assets through the "assets" array instead.
+   */
+  uri?: string;
+  /**
+   * @hidden
+   * @deprecated This field is deprecated and will be removed in SDK 48, you can access selected assets through the "assets" array instead.
+   */
+  assetId?: string | null;
+  /**
+   * @hidden
+   * @deprecated This field is deprecated and will be removed in SDK 48, you can access selected assets through the "assets" array instead.
+   */
+  width?: number;
+  /**
+   * @hidden
+   * @deprecated This field is deprecated and will be removed in SDK 48, you can access selected assets through the "assets" array instead.
+   */
+  height?: number;
+  /**
+   * @hidden
+   * @deprecated This field is deprecated and will be removed in SDK 48, you can access selected assets through the "assets" array instead.
+   */
+  type?: 'image' | 'video';
+  /**
+   * @hidden
+   * @deprecated This field is deprecated and will be removed in SDK 48, you can access selected assets through the "assets" array instead.
+   */
+  fileName?: string | null;
+  /**
+   * @hidden
+   * @deprecated This field is deprecated and will be removed in SDK 48, you can access selected assets through the "assets" array instead.
+   */
+  fileSize?: number;
+  /**
+   * @hidden
+   * @deprecated This field is deprecated and will be removed in SDK 48, you can access selected assets through the "assets" array instead.
+   */
+  exif?: Record<string, any> | null;
+  /**
+   * @hidden
+   * @deprecated This field is deprecated and will be removed in SDK 48, you can access selected assets through the "assets" array instead.
+   */
+  base64?: string | null;
+  /**
+   * @hidden
+   * @deprecated This field is deprecated and will be removed in SDK 48, you can access selected assets through the "assets" array instead.
+   */
+  duration?: number | null;
+} & (ImagePickerSuccessResult | ImagePickerCanceledResult);
+
 /**
- * An object returned when the pick action has been cancelled by the user.
+ * @hidden
```

Comment:
Why those types are hidden?

Reply:
- I don't want people to use them, `ImagePickerResult` should be enough for everyone
- Otherwise `ImagePickerResult` in the docs will be separated to two different types (tables) which is bad experience imho

Reply:
Will need to check this manually, since hiding those types could lead to broken links.

> Otherwise ImagePickerResult in the docs will be separated to two different types (tables) which is bad experience imho

This is an expected behavior since you are using types union or intersection. To avoid that I would recommend redefining the types, instead of attempting to workaround by hiding those.